### ISSUE TITLE
Adding customer model creates a shop migration

### ISSUE THREAD

**Issue by rubengrill** [awesto/django-shop#346]
I want to create a Customer extending shop's BaseCustomer (no custom fields yet, but they will come):

My models.py:
```python
from shop.models.customer import BaseCustomer

class Customer(BaseCustomer):
    pass
```

Don't see a difference to the doc: http://django-shop.readthedocs.io/en/latest/reference/customer-model.html#adding-the-customer-model-to-our-application

Though, when I run `makemigrations`, a new migration `0003_customer_orderpayment.py` is created in the shop app:

```python
# -*- coding: utf-8 -*-
from __future__ import unicode_literals

from django.db import migrations, models
import jsonfield.fields
from decimal import Decimal
import django.utils.timezone
from django.conf import settings


class Migration(migrations.Migration):

    dependencies = [
        ('email_auth', '0002_auto_20160327_1119'),
        ('shop', '0002_auto_20151016_1451'),
    ]

    operations = [
        migrations.CreateModel(
            name='Customer',
            fields=[
                ('user', models.OneToOneField(primary_key=True, serialize=False, to=settings.AUTH_USER_MODEL)),
                ('recognized', models.PositiveSmallIntegerField(default=0, help_text='Designates the state the customer is recognized as.', verbose_name='Recognized as', choices=[(0, 'Unrecognized'), (1, 'Guest'), (2, 'Registered')])),
                ('salutation', models.CharField(max_length=5, verbose_name='Salutation', choices=[('mrs', 'Mrs.'), ('mr', 'Mr.'), ('na', '(n/a)')])),
                ('last_access', models.DateTimeField(default=django.utils.timezone.now, verbose_name='Last accessed')),
                ('extra', jsonfield.fields.JSONField(default={}, verbose_name='Extra information about this customer', editable=False)),
            ],
        ),
        migrations.CreateModel(
            name='OrderPayment',
            fields=[
                ('id', models.AutoField(verbose_name='ID', serialize=False, auto_created=True, primary_key=True)),
                ('amount', models.DecimalField(default=Decimal('0'), help_text='How much was paid with this particular transfer.', max_digits=30, decimal_places=2)),
                ('transaction_id', models.CharField(help_text="The transaction processor's reference", max_length=255, verbose_name='Transaction ID')),
                ('created_at', models.DateTimeField(auto_now_add=True, verbose_name='Received at')),
                ('payment_method', models.CharField(help_text='The payment backend used to process the purchase', max_length=50, verbose_name='Payment method')),
            ],
            options={
                'verbose_name': 'Order payment',
                'verbose_name_plural': 'Order payments',
            },
        ),
    ]
```

Really don't get how this possible, as shop `models.py` don't import any model...

Used shop version: 0.9.1

**Comment by rfleschenberg** [comment]
Hi @rubengrill. Thank you for the report!

I suspect that the `OrderPayment` model migration is triggered by a setting, possibly by SHOP_ORDER_WORKFLOWS. I am not sure if we should consider this a bug or not.

**Comment by rfleschenberg** [comment]
Can you give us a link to your project, so we can reproduce things exactly?

**Comment by rfleschenberg** [comment]
It seems my assumption that this is triggered by a setting was wrong. I will wait for input from @jrief on this, since he knows the relevant code best.

**Comment by jrief** [comment]
@rubengrill this happens to a bug in how the DB MoneyField was managed. This bug has already been fixed, but not yet merged with the master.
Please try out the development version at https://github.com/jrief/django-shop . A ``./manage.py makemigrations`` will then create one more migration of the ``OrderPayment`` model, and convert the ``DecimalField`` into a ``shop.money.fields.MoneyField``.
Sorry for the inconvenience.

**Comment by rfleschenberg** [comment]
@rubengrill , by "shop app", do you mean `shop`, or do you mean your own app? If you mean `shop`, then this bug should be fixed in the current master, and the fix will be released in 0.9.2.

Otherwise, please provide a reproducible test case, e.g. a minimal example project.

Thanks!

**Comment by rubengrill** [comment]
Thanks for the quick response @rfleschenberg @jrief 
I tried the `develop` version of `https://github.com/jrief/django-shop`, but also here the migrations are created for the shop app.

Then I tried the `master` version of this repo (`-e git+https://github.com/awesto/django-shop@master#egg=django-shop
`), and also the migrations where added to the shop app.
By shop app I really mean the shop, not any of my apps.

What I've done so far:
`models.py`:

```python
from django.db import models
from shop.models.cart import BaseCart, BaseCartItem
from shop.models.customer import BaseCustomer
from shop.models.order import BaseOrder, BaseOrderItem
from shop.models.product import BaseProductManager, BaseProduct
from shop.money.fields import MoneyField


class Cart(BaseCart):
    pass


class CartItem(BaseCartItem):
    quantity = models.IntegerField()


class Customer(BaseCustomer):
    pass


class Order(BaseOrder):
    pass


class OrderItem(BaseOrderItem):
    quantity = models.IntegerField()


class Product(BaseProduct):
    slug = models.SlugField(unique=True)
    name = models.CharField(max_length=255)
    price = MoneyField()

    objects = BaseProductManager()
    lookup_fields = ('name__icontains',)

    @property
    def product_name(self):
        return self.name

    def get_price(self, request):
        return self.price

    def __unicode__(self):
        return self.slug
```

Within `INSTALLED_APPS`: `email_auth`, `shop`

Shop related settings:

```python
SHOP_APP_LABEL = 'shop'

SHOP_PAYMENT_BACKENDS = [
]

SHOP_SHIPPING_BACKENDS = [
]

SHOP_SHIPPING_FLAT_RATE = '0'

SHOP_DEFAULT_CURRENCY = 'EUR'
```

**Comment by rfleschenberg** [comment]
So far, I was not able to reproduce this. @rubengrill, can you please provide the full, exact commands and code to reproduce the issue? For example, in a git repository?

**Comment by rubengrill** [comment]
I found the problem when I was creating an example repository and copied the settings.
I set `SHOP_APP_LABEL` to `shop`, then migrations are created for the shop app.
When changing to my project's app name, the migrations are created correctly.

When assigning the value for this setting the first place I didn't figure out that this settings MUST be the name of my app, maybe this should be made more clear:
```
This label is required internally to configure the name of the database tables of used in the merchant’s implementation.
```
With the correct setting it works fine in 0.9.1
